sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/UnwrapCastInBinaryComparison.scala

```diff
@@ -156,29 +157,36 @@ object UnwrapCastInBinaryComparison extends Rule[LogicalPlan] {
         case lit @ Literal(null, _) => nullList += lit
         case lit @ NonNullLiteral(_, _) =>
           unwrapCast(EqualTo(in.value, lit)) match {
```

Comment:
now I start to think that creating `EqualTo` and reusing `unwrapCast` may not be a good idea here. Is it possible to pull the code of downcasting a value to a new function, and call that function here?

Reply:
added a new function `simplifyIn`.